tools/deviceGen/output/device_capabilities_traits_defaults.hlsl

```diff
@@ -0,0 +1,511 @@
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating16BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating16BitMixedSignednessAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitUnsignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating32BitMixedSignednessAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitUnsignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitSignedAccelerated, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(integerDotProductAccumulatingSaturating64BitMixedSignednessAccelerated, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(storageTexelBufferOffsetAlignmentBytes, uint64_t, nbl::hlsl::numeric_limits<uint64_t>::max);
+NBL_GENERATE_GET_OR_DEFAULT(uniformTexelBufferOffsetAlignmentBytes, uint64_t, nbl::hlsl::numeric_limits<uint64_t>::max);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxBufferSize, uint64_t, MinMaxSSBOSize);
+
+// Nabla Core Extensions
+NBL_GENERATE_GET_OR_DEFAULT(minImportedHostPointerAlignment, uint32_t, 0x1u << 31);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageFloat32AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32AtomicAdd, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(robustStorageBufferAccessSizeAlignment, uint64_t, 0x1ull << 63);
+NBL_GENERATE_GET_OR_DEFAULT(robustUniformBufferAccessSizeAlignment, uint64_t, 0x1ull << 63);
+
+// Extensions
+NBL_GENERATE_GET_OR_DEFAULT(shaderTrinaryMinmax, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderExplicitVertexParameter, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(gpuShaderHalfFloatAMD, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageLoadStoreLod, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(displayTiming, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxDiscardRectangles, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(primitiveOverestimationSizeBitPattern, uint32_t, asuint(0.f));
+NBL_GENERATE_GET_OR_DEFAULT(maxExtraPrimitiveOverestimationSizeBitPattern, uint32_t, asuint(0.f));
+NBL_GENERATE_GET_OR_DEFAULT(extraPrimitiveOverestimationSizeGranularityBitPattern, uint32_t, asuint(nbl::hlsl::numeric_limits<float>::infinity));
```

Comment:
`infinity` is already a uint IIRC

Reply:
Its generated like this because in the old file `extraPrimitiveOverestimationSizeGranularity` is a float. The script finds all floats and generates a BitPattern for them.

I can add an exception for infinity if you want.

Reply:
do you have `nbl::hlsl::numeric_limits<float>::infinity` in the JSON then?